airbyte-cdk/python/airbyte_cdk/sources/embedded/runner.py

```diff
@@ -0,0 +1,34 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+
+import logging
+from abc import ABC, abstractmethod
+from typing import Generic, Iterable, Optional
+
+from airbyte_cdk.connector import TConfig
+from airbyte_cdk.models import AirbyteCatalog, AirbyteMessage, AirbyteStateMessage, ConfiguredAirbyteCatalog
+from airbyte_cdk.sources.abstract_source import AbstractSource
+
+
+class SourceRunner(ABC, Generic[TConfig]):
+    @abstractmethod
+    def discover(self, config: TConfig) -> AirbyteCatalog:
+        pass
+
+    @abstractmethod
+    def read(self, config: TConfig, catalog: ConfiguredAirbyteCatalog, state: Optional[AirbyteStateMessage]) -> Iterable[AirbyteMessage]:
+        pass
+
+
+class CDKRunner(SourceRunner[TConfig]):
+    def __init__(self, source: AbstractSource, name: str):
```

Comment:
why does this have to be an AbstractSource / why not accept any Source?

Reply:
It can be any source, fixed this